caluma/workflow/tests/test_work_item.py

```diff
@@ -90,3 +98,109 @@ def test_complete_work_item_with_next(
 
     assert not result.errors
     snapshot.assert_match(result.data)
+
+
+@pytest.mark.parametrize(
+    "work_item__status,work_item__child_case", [(models.WorkItem.STATUS_READY, None)]
+)
+def test_complete_work_item_with_next_multiple_tasks(
+    db,
+    snapshot,
+    case,
+    work_item,
+    task,
+    task_factory,
+    task_flow_factory,
+    workflow,
+    schema_executor,
+):
+    task_next_1, task_next_2 = task_factory.create_batch(2)
+    task_flow = task_flow_factory(task=task, workflow=workflow)
+    task_flow.flow.next = f"['{task_next_1.slug}', '{task_next_2.slug}']|task"
+    task_flow.flow.save()
+
+    query = """
+        mutation CompleteWorkItem($input: CompleteWorkItemInput!) {
+          completeWorkItem(input: $input) {
+            workItem {
+              status
+              case {
+                status
+                workItems {
+                  edges {
+                    node {
+                      status
+                    }
+                  }
+                }
+              }
+            }
+            clientMutationId
+          }
+        }
+    """
+
+    inp = {"input": {"id": work_item.pk}}
+    result = schema_executor(query, variables=inp)
+
+    assert not result.errors
+    assert case.work_items.count() == 3
+    assert set(
+        case.work_items.values_list("task", flat=True).filter(
+            status=models.WorkItem.STATUS_READY
+        )
+    ) == {task_next_1.pk, task_next_2.pk}
+
+
+@pytest.mark.parametrize(
+    "work_item__status,work_item__child_case",
+    [(models.WorkItem.STATUS_COMPLETED, None)],
+)
+def test_complete_work_item_with_merge(
+    db,
+    snapshot,
+    case,
+    work_item,
+    work_item_factory,
+    task,
+    task_factory,
+    flow_factory,
+    task_flow_factory,
+    workflow,
+    schema_executor,
+):
+
+    work_item_1, work_item_2 = work_item_factory.create_batch(
+        2, status=models.WorkItem.STATUS_READY, child_case=None, case=case
+    )
+    task_next = task_factory()
+
+    flow = flow_factory(next=f"'{task_next.slug}'|task")
+    task_flow_factory(task=work_item_1.task, workflow=workflow, flow=flow)
+    task_flow_factory(task=work_item_2.task, workflow=workflow, flow=flow)
+
+    query = """
+        mutation CompleteWorkItem($input: CompleteWorkItemInput!) {
+          completeWorkItem(input: $input) {
+            workItem {
+              id
+            }
+          }
+        }
+    """
+
+    inp = {"input": {"id": work_item_1.pk}}
+    result = schema_executor(query, variables=inp)
+
+    assert not result.errors
+    assert case.work_items.filter(status=models.WorkItem.STATUS_READY).count() == 1
```

Comment:
I'd like to see some of the asserts before the to-be-tested operation. This helps the reader understand exactly what's supposed to happen, without inferring the state from the setup code. For example:
```python
assert ready_item_count == 2
operation_that_completes_an_item()
assert ready_item_count == 1
```

Reply:
Adjusted test case and also added some comments to hopefully make the test more comprehensible.